### PR TITLE
Add change log for June release, include the alpha release logs also.

### PR DESCRIPTION
**Pull request by lakhotiaharshit**
I have added change log of the backported PRs:
#2052 
#2051  
#2050 
#2049 
#2048 
#2047 
#2044 
#2041

Let me know if any other PR is missing. This is possible if it was not tagged with "backport needed" 

@jenshnielsen @astafan8  @GateBuilder

### CODE REVIEW

File: docs/changes/0.15.0.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Changelog for QCoDeS 0.15.0a1
```
Comment (jenshnielsen) [inline docs/changes/0.15.0.rst:1]: remove a1

**Comment by jenshnielsen** [comment]
You will also need to add a link to the file from the index. Please also build the docs locally and check that they render well

File: docs/changes/0.15.0.rst (hunk 1 above)
Comment (lakhotiaharshit) [reply docs/changes/0.15.0.rst]: done.

**Comment by astafan8** [comment]
all mentioned PRs were backported to v0.15.x branch. merging this once CI passes

**Pull request merged**